winml/lib/Api/LearningModelBinding.cpp

```diff
@@ -158,7 +158,7 @@ void LearningModelBinding::Bind(
 
   auto featureName = WinML::Strings::UTF8FromHString(name);
   std::tie(bindingName, binding_value, bindingType) = CreateBinding(featureName, value, properties);
-  Ort::Value ortValue = binding_value ? Ort::Value(binding_value) : Ort::Value(nullptr);
+  Ort::Value ortValue = Ort::Value(binding_value);
```

Comment:
>binding_value [](start = 35, length = 13)

what does this change?

Reply:
It's equivalent. It was using binding_value if it was non-null, or nullptr if it was null, so the comparison wasn't necessary.

Reply:
no.  it does different things unfortunately.   Ort::Value has an explicit contructor that takes nullptr.   this case bypasses an assert.  if you pass a var that has a "value of null" the constructor actually throws. fun :) 

---
In reply to: [356905801](https://github.com/microsoft/onnxruntime/pull/2635#discussion_r356905801) [](ancestors = 356905801)

Reply:
Base(T* p) : p_{p} {
    if (!p) throw Ort::Exception("Allocation failure", ORT_FAIL);
  }

Line 81 in onnxrutnime_cxx_api.h

---
In reply to: [356906174](https://github.com/microsoft/onnxruntime/pull/2635#discussion_r356906174) [](ancestors = 356906174,356905801)

Reply:
explicit Value(std::nullptr_t) {}


;ine 267

---
In reply to: [356906298](https://github.com/microsoft/onnxruntime/pull/2635#discussion_r356906298) [](ancestors = 356906298,356906174,356905801)

Reply:
I was just gonna comment about this :)

Reply:
Oh, my bad!